src/Libraries/CoreNodes/Data.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Autodesk.DesignScript.Runtime;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace DSCore
+{
+    public static class Data
+    {
+        /// <summary>
+        ///     Parse converts an arbitrary JSON string to a value. It is the opposite of JSON.Stringify.
+        /// </summary>
+        /// <param name="json">A JSON string</param>
+        /// <returns name="result">The result type depends on the content of the input string. The result type can be a primitive value (e.g. string, boolean, double), a List, or a Dictionary.</returns>
+        public static object ParseJSON(string json)
+        {
+            return ToNative(JToken.Parse(json));
+        }
+
+        private static object ToNative(JToken token)
+        {
+            switch (token.Type)
+            {
+                case JTokenType.Object:
+                    var obj = token as JObject;
+                    var dict = new Dictionary<string, object>();
+                    foreach (var kv in obj)
+                    {
+                        dict[kv.Key] = ToNative(kv.Value);
+                    }
+                    return dict;
+                case JTokenType.Array:
+                    var arr = token as JArray;
+                    return arr.Select(ToNative);
+                case JTokenType.Null:
+                    return null;
+                case JTokenType.Integer:
+                case JTokenType.Float:
+                case JTokenType.String:
+                case JTokenType.Boolean:
+                case JTokenType.Date:
+                case JTokenType.TimeSpan:
+                    return (token as JValue).Value;
+                case JTokenType.Guid:
+                case JTokenType.Uri:
+                    return (token as JValue).Value.ToString();
+                default:
+                    return null;
+            }
+        }
+
+        /// <summary>
+        ///     Stringify converts a list of arbitrary values to JSON. Replication can also be used to apply the operation over a list, producing a list of JSON strings.
```

Comment:
so is this actually true? Since this is an arbitraryRankArray I thought this never replicates?